asf_search/ASFProduct.py

```diff
@@ -48,11 +47,8 @@ def stack(self) -> UserList:
 
         return stack_from_product(self)
 
-    def centroid(self) -> (Iterable[float]):
+    def centroid(self) -> Point:
         """
         Finds the centroid of a product
-        Shamelessly lifted from https://stackoverflow.com/a/23021198 and https://stackoverflow.com/a/57183264
         """
-        arr = np.array(self.geometry['coordinates'][0])
-        length, dim = arr.shape
-        return [np.sum(arr[:, i]) / length for i in range(dim)]
+        return shape(self.geometry).centroid
```

Comment:
Certainly friendlier to read here.

Reply:
right? It almost makes me want to add a `.shape` or  `.to_shapely` (or something) method that returns `shape(self.geometry)` as well